packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit_webview_widget.dart

```diff
@@ -593,6 +603,16 @@ class WebKitWebViewPlatformController extends WebViewPlatformController {
 
     return value.toString();
   }
+
+  void _dispose() {
+    if (_progressObserverSet) {
+      _progressObserverSet = false;
+      // It is recommended by Apple's documentation that this should be called
+      // before the WebView is deallocated:
+      // https://developer.apple.com/documentation/objectivec/nsobject/1408054-removeobserver?language=objc
+      webView.removeObserver(webView, keyPath: 'estimatedProgress');
```

Comment:
I'm confused; the webview is its own observer?

If so I'm not clear what removing the observer here would accomplish, since there can't be a call to a dangling pointer if it's the same object.

Reply:
I just tested it and you are correct. There is no error when the WebView is deallocated. I went ahead and removed the `dispose` method.

Reply:
Ah, I think I originally misunderstood the documentation. It was recommending to remove the object that was added as an observer because notifications would still be sent to it if the observed object was still accessible. I understand it now.